diff --git a/chrome/browser/ui/views/toolbar/reload_button.cc b/chrome/browser/ui/views/toolbar/reload_button.cc
index fbabf7bf13cd4..538d57d990332 100644
--- a/chrome/browser/ui/views/toolbar/reload_button.cc
+++ b/chrome/browser/ui/views/toolbar/reload_button.cc
@@ -186,7 +186,7 @@ void ReloadButton::UpdateCachedTooltipText() {
 }
 
 bool ReloadButton::ShouldShowMenu() {
-  return is_dev_tools_connected_ && (visible_mode_ == Mode::kReload);
+  return visible_mode_ == Mode::kReload;
 }
 
 void ReloadButton::ShowDropDownMenu(ui::mojom::MenuSourceType source_type) {
