diff --git a/chrome/browser/ui/views/toolbar/reload_button.cc b/chrome/browser/ui/views/toolbar/reload_button.cc
index ac2fb7ae21769..326265446a3fb 100644
--- a/chrome/browser/ui/views/toolbar/reload_button.cc
+++ b/chrome/browser/ui/views/toolbar/reload_button.cc
@@ -195,7 +195,7 @@ void ReloadButton::UpdateCachedTooltipText() {
 }
 
 bool ReloadButton::ShouldShowMenu() {
-  return is_dev_tools_connected_ && (visible_mode_ == Mode::kReload);
+  return visible_mode_ == Mode::kReload;
 }
 
 void ReloadButton::ShowDropDownMenu(ui::mojom::MenuSourceType source_type) {
