diff --git a/third_party/libaom/BUILD.gn b/third_party/libaom/BUILD.gn
index d8dbb5b38859f..33e7684c204e0 100644
--- a/third_party/libaom/BUILD.gn
+++ b/third_party/libaom/BUILD.gn
@@ -229,6 +229,7 @@ if (current_cpu == "arm64" || current_cpu == "arm64e") {
     configs += [ "//build/config/compiler:no_chromium_code" ]
     configs += [ ":libaom_config" ]
     configs -= [ "//build/config/compiler:default_init_stack_vars" ]
+    configs -= [ "//build/config/compiler:thorium_simd_optimization" ]
     if (!is_win || is_clang) {
       cflags = [ "-march=armv8-a+crc" ]
     }
@@ -241,6 +242,7 @@ if (current_cpu == "arm64" || current_cpu == "arm64e") {
     configs += [ "//build/config/compiler:no_chromium_code" ]
     configs += [ ":libaom_config" ]
     configs -= [ "//build/config/compiler:default_init_stack_vars" ]
+    configs -= [ "//build/config/compiler:thorium_simd_optimization" ]
     if (!is_win || is_clang) {
       cflags = [ "-march=armv8.2-a+dotprod" ]
     }
@@ -255,6 +257,7 @@ if (current_cpu == "arm64" || current_cpu == "arm64e") {
     configs += [ "//build/config/compiler:no_chromium_code" ]
     configs += [ ":libaom_config" ]
     configs -= [ "//build/config/compiler:default_init_stack_vars" ]
+    configs -= [ "//build/config/compiler:thorium_simd_optimization" ]
     if (!is_win || is_clang) {
       cflags = [ "-march=armv8.2-a+dotprod+i8mm" ]
     }
@@ -272,6 +275,7 @@ if (current_cpu == "arm64" || current_cpu == "arm64e") {
       configs += [ "//build/config/compiler:no_chromium_code" ]
       configs += [ ":libaom_config" ]
       configs -= [ "//build/config/compiler:default_init_stack_vars" ]
+      configs -= [ "//build/config/compiler:thorium_simd_optimization" ]
       if (!is_win || is_clang) {
         cflags = [ "-march=armv8.2-a+dotprod+i8mm+sve" ]
       }
@@ -285,6 +289,7 @@ if (current_cpu == "arm64" || current_cpu == "arm64e") {
       configs += [ "//build/config/compiler:no_chromium_code" ]
       configs += [ ":libaom_config" ]
       configs -= [ "//build/config/compiler:default_init_stack_vars" ]
+      configs -= [ "//build/config/compiler:thorium_simd_optimization" ]
       if (!is_win || is_clang) {
         cflags = [ "-march=armv9-a+i8mm+sve2" ]
       }
diff --git a/third_party/libaom/cmake_update.sh b/third_party/libaom/cmake_update.sh
index b863f7ef73c27..d69bd50233c98 100755
--- a/third_party/libaom/cmake_update.sh
+++ b/third_party/libaom/cmake_update.sh
@@ -1,6 +1,6 @@
 #!/bin/bash
 #
-# Copyright 2018 The Chromium Authors
+# Copyright 2026 The Chromium Authors and Alex313031
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
@@ -22,6 +22,8 @@
 # - gcc-aarch64-linux-gnu
 # - g++-aarch64-linux-gnu
 # Toolchain for x86:
+# - lib32gcc-13-dev
+# - lib32stdc++-13-dev
 # - gcc-i686-linux-gnu
 # - g++-i686-linux-gnu
 # Alternatively: treat 32bit builds like Windows and manually tweak aom_config.h
@@ -178,14 +180,14 @@ egrep \
 
 reset_dirs linux/arm-neon
 gen_config_files linux/arm-neon \
-  "${toolchain}/armv7-linux-gcc.cmake -DCONFIG_RUNTIME_CPU_DETECT=0 \
+  "${toolchain}/armv7-linux-gcc.cmake -DCONFIG_RUNTIME_CPU_DETECT=0 -DENABLE_ARM_CRC32=0 -DENABLE_NEON_DOTPROD=0 -DENABLE_NEON_I8MM=0 \
    ${all_platforms}"
 
 reset_dirs linux/arm64-cpu-detect
 # Note clang is use to allow detection of SVE/SVE2; gcc as of version 13 is
 # missing the required arm_neon_sve_bridge.h header.
 gen_config_files linux/arm64-cpu-detect \
-  "${toolchain}/arm64-linux-clang.cmake ${all_platforms}"
+  "${toolchain}/arm64-linux-clang.cmake -DENABLE_ARM_CRC32=0 -DENABLE_NEON_DOTPROD=0 -DENABLE_NEON_I8MM=0 ${all_platforms}"
 
 # Generate linux configurations and modify for Windows.
 reset_dirs win/arm64-cpu-detect
@@ -200,7 +202,8 @@ convert_to_windows "${CFG}/win/arm64-cpu-detect/config/aom_config.h"
 
 update_readme
 
-git cl format > /dev/null \
-  || echo "ERROR: 'git cl format' failed. Please run 'git cl format' manually."
+# git cl format > /dev/null \
+#   || echo "WARNING: 'git cl format' failed. Please run 'git cl format' manually."
+echo "NOTE: 'You may want to run `git cl format` manually.'"
 
 cleanup
diff --git a/third_party/libvpx/BUILD.gn b/third_party/libvpx/BUILD.gn
index 3437424b46ef2..338595aed8ef6 100644
--- a/third_party/libvpx/BUILD.gn
+++ b/third_party/libvpx/BUILD.gn
@@ -313,6 +313,7 @@ if (current_cpu == "arm" || current_cpu == "arm64" || current_cpu == "arm64e") {
     configs -= [ "//build/config/compiler:chromium_code" ]
     configs += [ "//build/config/compiler:no_chromium_code" ]
     configs += [ ":libvpx_config" ]
+    configs -= [ "//build/config/compiler:thorium_simd_optimization" ]
     if (current_cpu == "arm") {
       configs -= [ "//build/config/compiler:compiler_arm_fpu" ]
       cflags = [ "-mfpu=neon" ]
@@ -340,6 +341,7 @@ if (current_cpu == "arm64" || current_cpu == "arm64e") {
     configs -= [ "//build/config/compiler:chromium_code" ]
     configs += [ "//build/config/compiler:no_chromium_code" ]
     configs += [ ":libvpx_config" ]
+    configs -= [ "//build/config/compiler:thorium_simd_optimization" ]
     if (!is_win || is_clang) {
       cflags = [ "-march=armv8.2-a+dotprod" ]
     }
@@ -357,6 +359,7 @@ if (current_cpu == "arm64" || current_cpu == "arm64e") {
     configs -= [ "//build/config/compiler:chromium_code" ]
     configs += [ "//build/config/compiler:no_chromium_code" ]
     configs += [ ":libvpx_config" ]
+    configs -= [ "//build/config/compiler:thorium_simd_optimization" ]
     if (!is_win || is_clang) {
       cflags = [ "-march=armv8.2-a+i8mm" ]
     }
@@ -378,6 +381,7 @@ if (current_cpu == "arm64" || current_cpu == "arm64e") {
       configs -= [ "//build/config/compiler:chromium_code" ]
       configs += [ "//build/config/compiler:no_chromium_code" ]
       configs += [ ":libvpx_config" ]
+      configs -= [ "//build/config/compiler:thorium_simd_optimization" ]
       cflags = [ "-march=armv8.2-a+dotprod+i8mm+sve" ]
       if (cpu_arch_full == "arm64") {
         sources = libvpx_srcs_arm64_sve
@@ -393,6 +397,7 @@ if (current_cpu == "arm64" || current_cpu == "arm64e") {
       configs -= [ "//build/config/compiler:chromium_code" ]
       configs += [ "//build/config/compiler:no_chromium_code" ]
       configs += [ ":libvpx_config" ]
+      configs -= [ "//build/config/compiler:thorium_simd_optimization" ]
       cflags = [ "-march=armv9-a+sve2" ]
       if (cpu_arch_full == "arm64") {
         sources = libvpx_srcs_arm64_sve2
@@ -562,6 +567,7 @@ static_library("libvpx") {
   }
   if (cpu_arch_full == "arm-neon-highbd" || cpu_arch_full == "arm-neon" ||
       current_cpu == "arm64" || current_cpu == "arm64e") {
+    configs -= [ "//build/config/compiler:thorium_simd_optimization" ]
     deps += [ ":libvpx_intrinsics_neon" ]
   }
   if (current_cpu == "arm64" || current_cpu == "arm64e") {
diff --git a/third_party/xnnpack/build_defs.gni b/third_party/xnnpack/build_defs.gni
index f12e9c7714e31..d3ac4c29feda1 100644
--- a/third_party/xnnpack/build_defs.gni
+++ b/third_party/xnnpack/build_defs.gni
@@ -16,7 +16,7 @@ xnn_defines = [
   "XNN_ENABLE_SSSE3=1",
   "XNN_ENABLE_SSE41=1",
   "XNN_ENABLE_ARM_FP16_SCALAR=0",
-  "XNN_ENABLE_ARM_FP16_VECTOR=0",
+  "XNN_ENABLE_ARM_FP16_VECTOR=1",
   "XNN_ENABLE_ARM_BF16=0",
 ]
 
