diff --git a/ffmpeg_generated.gni b/ffmpeg_generated.gni
index cbd0461e15..8d1bdb77f4 100644
--- a/ffmpeg_generated.gni
+++ b/ffmpeg_generated.gni
@@ -360,6 +360,7 @@ if ((current_cpu == "x64" && ffmpeg_branding == "Chrome") ||
     "libavcodec/x86/h26x/h2656dsp.c"
   ]
 
+if (!(is_android && current_cpu == "x86")) {
   ffmpeg_asm_sources += [
     "libavcodec/x86/bswapdsp.asm",
     "libavcodec/x86/hevc/add_res.asm",
@@ -371,6 +372,7 @@ if ((current_cpu == "x64" && ffmpeg_branding == "Chrome") ||
     "libavcodec/x86/hevc/sao_10bit.asm",
     "libavcodec/x86/h26x/h2656_inter.asm"
   ]
+}
 
   ffmpeg_c_sources += [
     "libavcodec/x86/autorename_libavcodec_x86_aacpsdsp_init.c",
