diff --git a/chrome/browser/ui/views/tabs/tab_strip.cc b/chrome/browser/ui/views/tabs/tab_strip.cc
index b90c536bbf25a1eb2635431453f104d4b36e2d32..202866d75a82b2795f321eb840361d5ab6fa282b 100644
--- a/chrome/browser/ui/views/tabs/tab_strip.cc
+++ b/chrome/browser/ui/views/tabs/tab_strip.cc
@@ -1388,8 +1388,17 @@ void TabStrip::OnSplitContentsChanged(const std::vector<int>& split_indices) {
 }
 
 bool TabStrip::ShouldDrawStrokes() const {
+  // Don't want to have to run a full feature query every time this function is
+  // called.
+  static const bool force_disable_tab_outlines =
+      base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "force-disable-tab-outlines");
 #if BUILDFLAG(IS_CHROMEOS)
-  return false;
+  if (force_disable_tab_outlines) {
+    return false;
+  } else {
+    return true;
+  }
 #else   // BUILDFLAG(IS_CHROMEOS)
 
   // If the controller says we can't draw strokes, don't.
@@ -1408,7 +1417,7 @@ bool TabStrip::ShouldDrawStrokes() const {
   // The Tabstrip in the refreshed style does not meet the contrast ratio
   // requirements listed below but does not have strokes for Tabs or the bottom
   // border.
-  if (!using_system_theme) {
+  if (!using_system_theme && force_disable_tab_outlines) {
     return false;
   }
 
@@ -1425,7 +1434,14 @@ bool TabStrip::ShouldDrawStrokes() const {
       controller_->GetFrameColor(BrowserFrameActiveState::kActive);
   const float contrast_ratio =
       color_utils::GetContrastRatio(background_color, frame_color);
-  return contrast_ratio < kMinimumContrastRatioForOutlines;
+  if (contrast_ratio < kMinimumContrastRatioForOutlines) {
+    return true;
+  }
+
+  if (!force_disable_tab_outlines) {
+    return true;
+  }
+  return false;
 #endif  // BUILDFLAG(IS_CHROMEOS)
 }
 
