diff --git a/BUILD.gn b/BUILD.gn
index 2288e24a..9c3b6792 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -209,6 +210,7 @@ static_library("libyuv_internal") {
 
 if (libyuv_use_neon) {
   static_library("libyuv_neon") {
+    configs -= [ "//build/config/compiler:thorium_simd_optimization" ]
     sources = [
       # ARM Source Files
       "source/compare_neon.cc",
@@ -237,6 +239,7 @@ if (libyuv_use_neon) {
 
 if (libyuv_use_sve) {
   static_library("libyuv_sve") {
+    configs -= [ "//build/config/compiler:thorium_simd_optimization" ]
     sources = [ "source/row_sve.cc" ]
 
     deps = [ ":libyuv_internal" ]
@@ -253,6 +256,7 @@ if (libyuv_use_sve) {
 
 if (libyuv_use_sme) {
   static_library("libyuv_sme") {
+    configs -= [ "//build/config/compiler:thorium_simd_optimization" ]
     sources = [
       "source/rotate_sme.cc",
       "source/row_sme.cc",
@@ -273,6 +277,7 @@ if (libyuv_use_sme) {
 
 if (libyuv_use_lsx) {
   static_library("libyuv_lsx") {
+    configs -= [ "//build/config/compiler:thorium_simd_optimization" ]
     sources = [
       # LSX Source Files
       "source/rotate_lsx.cc",
@@ -293,6 +298,7 @@ if (libyuv_use_lsx) {
 
 if (libyuv_use_lasx) {
   static_library("libyuv_lasx") {
+    configs -= [ "//build/config/compiler:thorium_simd_optimization" ]
     sources = [
       # LASX Source Files
       "source/row_lasx.cc",
