diff --git a/chrome/browser/net/default_dns_over_https_config_source.cc b/chrome/browser/net/default_dns_over_https_config_source.cc
index e5a2aadcc85f6..3ce7e0d2088fd 100644
--- a/chrome/browser/net/default_dns_over_https_config_source.cc
+++ b/chrome/browser/net/default_dns_over_https_config_source.cc
@@ -1,4 +1,4 @@
-// Copyright 2024 The Chromium Authors
+// Copyright 2026 The Chromium Authors and Alex313031
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -29,7 +29,7 @@ DefaultDnsOverHttpsConfigSource::DefaultDnsOverHttpsConfigSource(
   if (set_up_pref_defaults) {
     local_state->SetDefaultPrefValue(prefs::kDnsOverHttpsMode,
                                      base::Value(SecureDnsConfig::ModeToString(
-                                         net::SecureDnsMode::kAutomatic)));
+                                         net::SecureDnsMode::kSecure)));
   }
 }
 
diff --git a/chrome/browser/net/stub_resolver_config_reader.cc b/chrome/browser/net/stub_resolver_config_reader.cc
index 3d9047c1cf558..49bee0a177409 100644
--- a/chrome/browser/net/stub_resolver_config_reader.cc
+++ b/chrome/browser/net/stub_resolver_config_reader.cc
@@ -1,4 +1,4 @@
-// Copyright 2020 The Chromium Authors
+// Copyright 2026 The Chromium Authors and Alex313031
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -238,42 +238,11 @@ void StubResolverConfigReader::UpdateNetworkService(bool record_metrics) {
 }
 
 bool StubResolverConfigReader::ShouldDisableDohForManaged() {
-// This function ignores cloud policies which are loaded on a per-profile basis.
-#if BUILDFLAG(IS_ANDROID)
-  // Check for MDM/management/owner apps. android_has_owner_ is true if either a
-  // device or policy owner app is discovered by
-  // GetAndroidEnterpriseInfoState(). If android_has_owner_ is nullopt, take a
-  // value of false so that we don't disable DoH during the async check.
-
-  // Because Android policies can only be loaded with owner apps this is
-  // sufficient to check for the prescences of policies as well.
-  if (android_has_owner_.value_or(false))
-    return true;
-#elif BUILDFLAG(IS_WIN)
-  // TODO(crbug.com/40229843): What is the correct function to use here? (This
-  // may or may not obsolete the following TODO)
-  // TODO(crbug.com/40223626): For legacy compatibility, this uses
-  // IsEnterpriseDevice() which effectively equates to a domain join check.
-  // Consider whether this should use IsManagedDevice() instead.
-  if (base::win::IsEnrolledToDomain())
-    return true;
-#endif
-#if !BUILDFLAG(IS_ANDROID) && !BUILDFLAG(IS_CHROMEOS)
-  if (g_browser_process->browser_policy_connector()->HasMachineLevelPolicies())
-    return true;
-#endif
   return false;
 }
 
 bool StubResolverConfigReader::ShouldDisableDohForParentalControls() {
-  if (parental_controls_testing_override_.has_value())
-    return parental_controls_testing_override_.value();
-
-#if BUILDFLAG(IS_WIN)
-  return ShouldDisableDohForWindowsParentalControls();
-#else
   return false;
-#endif
 }
 
 void StubResolverConfigReader::SetOverrideDnsOverHttpsConfigSource(
